### ISSUE TITLE
Plugin doesn't follow the same module-loader search paths

### ISSUE THREAD

**Issue by richardspence** [LeDDGroup/typescript-transform-paths#30]
## Summary
The plugin treats the paths in the tsconfig as gospel for adjusting the resolution path.  It should instead use it as the primary search folder for determining where the module is located before falling back to the default module resolution strategies.  

## Repro steps
tsconfig:
```json
    "baseUrl": ".",
    "paths": {
            "*": ["*"]
        },
```
foo/IFoo
```ts
export interface IFoo {}
```
foo/File2
```ts
import {IFoo} from './IFoo';
```

## Expected
The plugin to not transform the relative imports (since they aren't available of the root).  Paths should be performed first and only applied if the module can be loaded from that location.  Then it should fallback to the module resolution strategy defined for the project https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies

## Actual
foo/File2.d.ts
```ts
import {IFoo} from '../IFoo';
```

**Comment by richardspence** [comment]
The fix should be to adjust `bindModuleToFile` to only return if the `targetFile[.ts|.js|.d.ts|.json|.jsx|.tsx]` exists on the location.  `JSON` and `JSX` or `TSX` only if the applicable compiler options are turned on.

**Comment by danielpza** [comment]
This could be tricky, as those are not the only extensions a module could have, eg it could be `.css`:
https://webpack.js.org/guides/typescript/#importing-other-assets

**Comment by richardspence** [comment]
Yup and you could have something like this as well:
```ts
declare module 'css!*' {
   var e: string;
   export default e;
}

...
import myCss from 'css!foo/myCss';
```

One option is to ignore the additional loaders like `css!`

For the other extensions you could make it configurable to add/replace/remove

**Comment by danielpza** [comment]
Hi @richardspence, I reread the tsconfig docs and yes, the only extensions allowed are `.ts`, `.d.ts`, `.jsx`, etc...

I release a fix, please upgrade to *v1.1.10*.
